Application/session_manager.py

```diff
@@ -0,0 +1,173 @@
+from PySide6.QtCore import Slot, QSettings
+from PySide6.QtGui import QCloseEvent
+
+from Models.session_entity import SessionEntity
+
+
+class SessionManager:
+    """
+    SessionManager is a manager for every session that takes in data
+    sent from the controller from the view and puts it in QSettings
+    and our session entity.
+    """
+    def __init__(self):
+        """
+        Constructor - loads our session entity
+        """
+        self.session_entity = SessionEntity()
+
+    def set_session_id(self, session_id):
+        """
+        Sets the session entity session id.
+
+        Parameters:
+            session id
+        """
+        self.session_entity.session_id = session_id
+
+    def set_table_name(self, table_name):
+        """
+        Sets the session entity table name.
+
+        Parameters:
+            table name
+        """
+        self.session_entity.table_name = table_name
+
+    def set_table_rows(self, row_count):
+        """
+        Sets the session entity row count
+
+        Parameters:
+            table row count
+        """
+        self.session_entity.table_row = row_count
+
+    def set_table_cols(self, col_count):
+        """
+        Sets the session entity column count
+
+        Parameters:
+            table column count
+        """
+        self.session_entity.table_col = col_count
+
+    def set_table_headers(self, headers):
+        """
+        Sets the session entity table headers
+
+        Parameters:
+            table header list
+        """
+        self.session_entity.table_headers = headers
+
+    def set_table_data(self, data):
+        """
+        Sets the session entity table data
+
+        Parameters:
+            table data 2D list
+        """
+        self.session_entity.table_data = data
+
+    def write_to_settings(self):
+        """
+        Takes the data in our session entity and writes it to QSettings
+        saved under a key of the session id name.
+        """
+        settings = QSettings()
+
+        settings.beginGroup(self.session_entity.session_id)
+        settings.beginGroup("encoding-table-panel")  # creates bin within session_id bin
+
+        settings.setValue("title", self.session_entity.table_name)
+
+        # Need to leave the bin.
```

Comment:
The comment here might be unnecessary. If we wish to keep it, we should at least make it in-line, where it doesn't take up as much space in the method.